Bottom: 4cc44fca8f0014d48b531c9cc0834c9618490654
Top:    e67d88d17cb26ea5305c4b691d21100a48de6508
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-08-02 13:19:02 +0800

tuntap: flush xdp map in the caller

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 3f23e2c..f9d3d29 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1715,6 +1715,9 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 		act = tun_do_xdp(tun, tfile, xdp_prog, &xdp, &err);
 		if (err)
 			goto err_xdp;
+
+		if (act == XDP_REDIRECT)
+			xdp_do_flush_map();
 		if (act != XDP_PASS)
 			goto out;
